### ISSUE TITLE
Error after deleting a header in a development Pod

### ISSUE THREAD

**Issue by pnc** [CocoaPods/CocoaPods#3914]
### Command

```
/Users/eleos/.rbenv/versions/2.2.2/bin/pod install
```

### Report

* What did you do?

If you run `pod install`, delete a header file from a development pod, and run `pod install` again, CocoaPods crashes.

* What did you expect to happen?

`pod install` should run normally, deleting the header from the `Pods/` directory as needed.

* What happened instead?

CocoaPods crashes trying to `chmod` the header's broken link.

### Stack

```
   CocoaPods : 0.38.2
        Ruby : ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin14]
    RubyGems : 2.4.5
        Host : Mac OS X 10.10.3 (14D136)
       Xcode : 6.3.2 (6D2105)
         Git : git version 2.2.1
Ruby lib dir : /Users/eleos/.rbenv/versions/2.2.2/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ e651ffcab678d6ad932a6b566681a8b13eff3ab9
```

### Plugins

```
cocoapods-plugins : 0.4.2
cocoapods-stats   : 0.5.3
cocoapods-trunk   : 0.6.1
cocoapods-try     : 0.4.5
```

### Podfile

```ruby
source 'https://github.com/CocoaPods/Specs.git'
use_frameworks!

target 'chmod_Example', :exclusive => true do
  pod "chmod", :path => "../"
end

target 'chmod_Tests', :exclusive => true do
  pod "chmod", :path => "../"

  
end
```

### Error

```
Errno::ENOENT - No such file or directory @ rb_file_s_stat - /Users/eleos/cocoapods-duplicate/chmod/Example/Pods/Headers/Private/chmod/Example.h
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/2.2.0/fileutils.rb:904:in `stat'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/2.2.0/fileutils.rb:904:in `symbolic_modes_to_i'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/2.2.0/fileutils.rb:951:in `fu_mode'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/2.2.0/fileutils.rb:1025:in `block (2 levels) in chmod_R'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/2.2.0/fileutils.rb:1477:in `preorder_traverse'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/2.2.0/fileutils.rb:1023:in `block in chmod_R'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/2.2.0/fileutils.rb:1022:in `each'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/2.2.0/fileutils.rb:1022:in `chmod_R'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/cocoapods-0.38.2/lib/cocoapods/installer.rb:117:in `block in prepare'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/cocoapods-0.38.2/lib/cocoapods/user_interface.rb:140:in `message'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/cocoapods-0.38.2/lib/cocoapods/installer.rb:116:in `prepare'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/cocoapods-0.38.2/lib/cocoapods/installer.rb:103:in `install!'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/cocoapods-0.38.2/lib/cocoapods/command/project.rb:71:in `run_install_with_update'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/cocoapods-0.38.2/lib/cocoapods/command/project.rb:101:in `run'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/claide-0.9.1/lib/claide/command.rb:312:in `run'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/cocoapods-0.38.2/lib/cocoapods/command.rb:48:in `run'
/Users/eleos/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/cocoapods-0.38.2/bin/pod:44:in `<top (required)>'
/Users/eleos/.rbenv/versions/2.2.2/bin/pod:23:in `load'
/Users/eleos/.rbenv/versions/2.2.2/bin/pod:23:in `<main>'
```

You can reproduce this by doing:

    pod lib create chmod
    cd chmod
    touch Pod/Classes/Example.h
    cd Example
    pod install
    rm ../Pod/Classes/Example.h
    pod install

**Comment by oronbz** [comment]
I'm also experiencing this when removing any header file

**Comment by suculent** [comment]
I had the same issue. After deleting whole ./Pods/ directory that actually contained symlinks to deleted headers, the ’pod install’ command stopped failing.

**Comment by segiddins** [comment]
I have an open PR that fixes this.

**Comment by segiddins** [comment]
This should now be fixed on `master`.

**Comment by ewhite-dev** [comment]
I am using cocoapods 38.2 , how do I fix this without upgrading my cocoapods version? (all higher versions do not allow me to archive correctly)

**Comment by segiddins** [comment]
@ewhite-dev probably find the PR that fixed it and try cherry picking it to 0.38, but your best bet will be to try and update.
